diff --git Cabal2Ebuild.hs Cabal2Ebuild.hs
index e3a83a3..1a0526f 100644
--- Cabal2Ebuild.hs
+++ Cabal2Ebuild.hs
@@ -53,7 +53,7 @@ cabal2ebuild pkg = Portage.ebuildTemplate {
     E.my_pn           = if any isUpper cabalPkgName then Just cabalPkgName else Nothing,
     E.features        = E.features E.ebuildTemplate
                    ++ (if hasExe then ["bin"] else [])
-                   ++ maybe [] (const (["lib","profile","haddock","hoogle"]
+                   ++ maybe [] (const (["lib","profile","haddock"]
                         ++ if cabalPkgName == "hscolour" then [] else ["hscolour"])
                         ) (Cabal.library pkg) -- hscolour can't colour its own sources
   } where
